File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int leftContainer = 0;
        int rightContainer = height.length - 1;
        int maxArea = 0;
        while(leftContainer < rightContainer)
        {
            maxArea = Math.max(maxArea, Math.min(height[leftContainer], height[rightContainer]) * (rightContainer - leftContainer));
            if(height[rightContainer] < height[leftContainer])
            {
                rightContainer--;
            }
            else
            {
                leftContainer++;
            }
        }
        return maxArea;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int maxArea = 0;
        for(int i=n-1;i>0;i--)
        {
            int j = 0;
            while(j<i && height[j]<height[i])
            {
                j++;
            }
            if(j!=i)
            {
                maxArea = max(maxArea, height[i] * (i-j));  
            }
        }
        
        for(int i=0;i<n;i++)
        {
            int j = n-1;
            while(j>i && height[j]<height[i])
            {
                j--;
            }
            if(j!=i)
            {
                maxArea = max(maxArea, height[i] * (j-i));  
            }
        }
        return maxArea;
    }
};